### ISSUE TITLE
OOD performance degrades 1.0–7.4x vs IID when evaluating across chemical series

### ISSUE THREAD

**Issue by srijitseal** [srijitseal/polaris#11]
## Summary

Training XGBoost on the largest chemical series (Butina cluster 0, n=2,578) and evaluating on held-out molecules from the same series (IID) vs a different series (cluster 2, n=1,306, OOD) shows consistent and often dramatic performance degradation. This is the example (paper Fig 7) — it works because this dataset uniquely provides both temporal ordering (for time-split within a series) and chemical series structure (for OOD evaluation across series). The result is intuitive but quantifying it requires dataset properties that public benchmarks almost never have.

## Method

1. **Select two largest Butina clusters** (cutoff 0.7, from notebook 2.01): cluster 0 (n=2,578) and cluster 2 (n=1,306). These represent two distinct chemical series from the Expansion Tx drug discovery campaigns.
2. **Time-split cluster 0** using ordinal molecule index (E-XXXXXXX naming encodes temporal order): earlier 80% = train (n=2,062), later 20% = IID validation (n=516). This mimics how a medicinal chemist would deploy a model: train on past compounds, predict new ones in the same series.
3. **Cluster 2 = OOD test set** (n=1,306). This mimics predicting an entirely different chemical series — a harder but common real-world scenario (e.g., hit-to-lead on a new scaffold).
4. **Train default XGBoost** (`XGBRegressor(random_state=42)`, no hyperparameter tuning) per endpoint on ECFP4 (2048-bit) + full RDKit 2D descriptors (~200), with dimorphite_dl protonation at assay-relevant pH. No tuning is intentional — this is a case study about distribution shift, not model optimization.
5. **Compare squared error distributions** between IID and OOD predictions. Log-transform applied to all endpoints except LogD, matching the competition evaluation protocol.

### Split characterization

| Set | n | Cluster | 1-NN median | 1-NN mean |
|-----|---|---------|-------------|-----------|
| Train | 2,062 | 0 (largest) | — | — |
| IID val | 516 | 0 (time-split, later 20%) | 0.282 | 0.295 |
| OOD test | 1,306 | 2 (second largest) | 0.761 | 0.761 |

The distance distributions have near-zero overlap — IID molecules are structurally close to training data (same chemical series, Tanimoto distance ~0.28), while OOD molecules are far away (different series, ~0.76). This clean separation is what makes the comparison compelling: any performance difference can be attributed to distribution shift rather than noise.

### Per-endpoint performance

| Endpoint | IID R² | OOD R² | IID Spearman | OOD Spearman | IID RAE | OOD RAE | IID median SE | OOD median SE | SE fold-change |
|----------|--------|--------|-------------|-------------|---------|---------|--------------|--------------|----------------|
| MLM CLint | -0.06 | -2.15 | 0.512 | 0.025 | 0.983 | 1.863 | 0.085 | 1.094 | **12.9x** |
| KSOL | 0.24 | -0.02 | 0.501 | 0.269 | 0.824 | 0.947 | 0.039 | 0.393 | **10.0x** |
| MPPB | 0.08 | -0.02 | 0.421 | 0.108 | 0.920 | 1.015 | 0.016 | 0.112 | **7.1x** |
| HLM CLint | -0.61 | -0.17 | 0.305 | 0.060 | 1.160 | 1.114 | 0.048 | 0.257 | **5.3x** |
| LogD | 0.53 | 0.09 | 0.750 | 0.315 | 0.637 | 0.960 | 0.138 | 0.596 | **4.3x** |
| Caco-2 Efflux | 0.48 | -0.37 | 0.737 | -0.005 | 0.608 | 1.140 | 0.052 | 0.136 | **2.6x** |
| MBPB | 0.14 | -0.17 | 0.439 | 0.414 | 0.919 | 1.174 | 0.034 | 0.086 | **2.5x** |
| Caco-2 Papp A>B | 0.47 | -0.15 | 0.697 | 0.126 | 0.667 | 1.081 | 0.053 | 0.063 | **1.2x** |

R² goes negative on OOD for 7 of 8 endpoints — the model is worse than predicting the mean when deployed on a new chemical series. Only LogD retains positive (but low) R² on OOD, likely because LogD is the most well-covered endpoint (96% of molecules) and is driven by global physicochemical properties rather than series-specific SAR.

Spearman ρ collapses on OOD: from 0.50–0.75 (IID) to near-zero or negative (OOD) for most endpoints. RAE exceeds 1.0 on OOD for 6 of 8 endpoints, meaning the model is worse than predicting the mean — a baseline that requires no training at all.

The clearance endpoints (MLM CLint, HLM CLint) show the largest degradation, consistent with metabolic stability being highly sensitive to specific structural features within a chemical series. Caco-2 Papp A>B shows the smallest gap (1.2x), suggesting permeability may generalize better across series — possibly because it depends more on bulk molecular properties (size, polarity) than on specific pharmacophoric features.

MGMB was skipped due to insufficient data in both clusters (only 431 molecules total, sparsely distributed).

### Why this matters

Standard IID evaluation (random split or even time-split within a single series) dramatically overestimates real-world performance. When a model is deployed on a genuinely new chemical series — the common scenario in hit identification — performance collapses. This analysis is only possible because the Expansion Tx dataset provides:
- **Chemical series structure**: Butina clustering identifies coherent series with hundreds of members
- **Temporal ordering**: Ordinal molecule naming enables realistic time-split within a series

Public ADMET benchmarks typically lack both properties, making this kind of IID vs OOD decomposition impossible.

## Plots

<img width="1178" height="728" alt="Image" src="https://github.com/user-attachments/assets/97d41215-cf2b-4c5f-b2cc-53a70bc9e0b9" />
<img width="1478" height="727" alt="Image" src="https://github.com/user-attachments/assets/9876d141-2d73-4a4c-a0f3-cfb48914d50d" />
<img width="1478" height="727" alt="Image" src="https://github.com/user-attachments/assets/d4532180-4533-409b-9719-e905331ebd6b" />
<img width="1478" height="727" alt="Image" src="https://github.com/user-attachments/assets/b07e7b61-72e6-4458-8041-7261bd05f4af" />
<img width="1478" height="727" alt="Image" src="https://github.com/user-attachments/assets/83ebdbf4-1743-45ef-ba13-fa18a1fdbef2" />
<img width="1478" height="727" alt="Image" src="https://github.com/user-attachments/assets/2bd7b571-52a0-487e-ae91-0a83ae8888d5" />
<img width="1478" height="727" alt="Image" src="https://github.com/user-attachments/assets/e583dcfa-63d8-4ee3-96f9-199e9d829ce1" />
<img width="2677" height="1824" alt="Image" src="https://github.com/user-attachments/assets/d7b648f7-c9f8-4ba6-bf0c-267eb4aa3336" />

## Reproduce

```bash
pixi run -e cheminformatics python notebooks/2.09-seal-iid-vs-ood-series.py
```

## Source

`notebooks/2.09-seal-iid-vs-ood-series.py`

**Comment by srijitseal** [comment]
## CheMeleon update (2.09 IID vs OOD)

Re-ran the IID-vs-OOD evaluation with CheMeleon. Same setup: train on largest Butina cluster (n=2,572) split temporally 80/20 for IID validation, OOD test on second-largest cluster (n=1,301).

**Verdict: trend matches XGBoost — substantial OOD degradation persists. But the foundation model does *not* close the IID–OOD gap; on 6/8 endpoints OOD R² is *worse* than XGBoost. Foundation model lifts IID baselines and helps LogD specifically; everything else either equals or regresses.**

### IID metrics

| Endpoint | R² IID XGB | R² IID CM | RAE IID XGB | RAE IID CM |
|----------|-----------|----------|-------------|------------|
| LogD | 0.645 | 0.640 | 0.575 | 0.573 |
| KSOL | 0.291 | 0.157 | 0.835 | 0.675 |
| HLM CLint | −0.029 | −0.645 | 0.923 | 1.189 |
| MLM CLint | 0.107 | 0.142 | 0.915 | 0.907 |
| Caco-2 Papp | 0.448 | 0.448 | 0.691 | 0.662 |
| Caco-2 Efflux | 0.456 | 0.402 | 0.621 | 0.659 |
| MPPB | 0.162 | 0.405 | 0.927 | 0.801 |
| MBPB | 0.237 | 0.417 | 0.843 | 0.700 |

### OOD metrics

| Endpoint | R² OOD XGB | R² OOD CM | RAE OOD XGB | RAE OOD CM |
|----------|-----------|----------|-------------|------------|
| LogD | +0.055 | **+0.444** | 0.982 | 0.784 |
| KSOL | +0.119 | −0.560 | 0.902 | 1.020 |
| HLM CLint | −0.012 | −0.096 | 1.003 | 1.079 |
| MLM CLint | −0.822 | −1.113 | 1.340 | 1.482 |
| Caco-2 Papp | −0.041 | −0.320 | 1.034 | 1.150 |
| Caco-2 Efflux | −0.005 | −0.062 | 1.002 | 1.042 |
| MPPB | +0.146 | +0.172 | 0.933 | 0.881 |
| MBPB | −0.488 | −1.231 | 1.340 | 1.697 |

R² goes negative OOD for **6/8 endpoints under CheMeleon** (vs 5/8 under XGBoost). The only endpoint where CheMeleon meaningfully improves OOD generalization is LogD (R² 0.05 → 0.44).

### Median squared-error fold change (OOD/IID)

| Endpoint | XGB | CM |
|----------|-----|-----|
| Caco-2 Efflux | 2.15× | 1.97× |
| Caco-2 Papp | 0.98× | 1.34× |
| HLM CLint | 5.16× | 4.62× |
| KSOL | 6.55× | 46.93× |
| LogD | 5.11× | 4.40× |
| MBPB | 5.09× | 19.00× |
| MLM CLint | 6.35× | 8.06× |
| MPPB | 5.55× | 2.79× |

KSOL and MBPB show extreme fold-change for CheMeleon because IID median SE is very small (the model fits the within-series IID set tightly) while OOD median SE is comparable to XGBoost — making the *gap* worse, not the OOD performance better. This is the classic "great IID, bad OOD" failure: tighter within-series fit → larger relative cliff at the series boundary.

### Trend match

Yes, qualitatively. The headline finding — *models trained on one chemical series degrade substantially when applied to a different series, with R² often going negative* — replicates and is in fact strengthened by CheMeleon. The foundation model does not solve cross-series generalization; for most endpoints it makes the IID/OOD divergence more pronounced.

### Source

- `data/processed/2.09-seal-iid-vs-ood-series/chemeleon/summary_metrics.csv`
- `data/processed/2.09-seal-iid-vs-ood-series/combined/`
- Reproduce: `pixi run -e cheminformatics python notebooks/2.09-seal-iid-vs-ood-series.py --model chemeleon`

**Comment by srijitseal** [comment]
## Refreshed numbers (XGBoost rerun + CheMeleon)

Issue body re-synced against `data/processed/2.09-seal-iid-vs-ood-series/xgboost/`. Headline drift:

- Title corrected: "1.0–7.4×" → **"up to 6.6×"** in median SE fold-change
- R² goes negative OOD on **5/8 endpoints** (HLM, MLM, Caco-2 Papp, Caco-2 Efflux, MBPB), not 3/8 as originally stated
- LogD OOD R²: 0.11 → **0.06**; HLM CLint IID R²: −0.12 → **−0.03**; KSOL OOD R²: 0.15 → **0.12**
- Median SE fold change: KSOL 6.8× → **6.5×**; MLM 7.4× → **6.4×**; LogD 4.9× → **5.1×**; MBPB 5.6× → **5.1×**; Caco-2 Efflux 2.1× → **2.2×**

The "default XGBoost" comparator numbers in the original body have been removed since no current artifact backs them.

CheMeleon comparison (R² negative OOD on 6/8 endpoints, helps LogD only, widens gap on most others) stands — see #issuecomment-4376602653.
